src/main/java/ensa/ebanking/accountservice/Services/AgentService.java

```diff
@@ -57,24 +62,50 @@ public List<ClientProfile> invalideAccounts () {
         return clientProfileDAO.findClientProfileByAccountStatus(AccountStatus.INACTIVE);
     }
 
-    public ClientProfile validAccount (String json){
+    public ClientProfile validAccount (ClientProfile clientProfile){
         try {
-            int id = (Integer) new JSONObject(json).get("id");
-            ClientProfile profile = clientProfileDAO.findClientProfileById(id);
-            profile.setAccountStatus(AccountStatus.ACTIVE);
-            return clientProfileDAO.save(profile);
+            Optional<ClientProfile> optionalClientProfile = clientProfileDAO.findById(clientProfile.getId());
+            if (optionalClientProfile.isPresent()) {
+                ClientProfile client = optionalClientProfile.get();
+                String email = client.getEmail();
+                client.setAccountStatus(AccountStatus.ACTIVE);
+                String subject = "Validation de votre compte";
+                String body = "Nous sommes ravis de vous informer que votre demande a été validée. Vous pouvez désormais " +
+                        "bénéficier des services de ENSAPAY.";
+                this.emailSenderService.sendEmail(email, subject, body);
+                return clientProfileDAO.save(client);
+            }
+            else {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+            }
 
         } catch (Exception e) {
-            System.out.println("account not found");
+            e.printStackTrace();
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Internal Server Error doesn't suit the purpose here, 500 means an unexpected error has happened.

Reply:
Which error do you suggest?

Reply:
I suggest 404 not found

Reply:
I sent that one when the Optional<ClientProfile> is null, i.e there's no user in the database with that id. This one was when any other error was raised. I can make this a 404 error if you want.

Reply:
fga3 l a7wal 404 hsn mn 500

Reply:
Ok will change it and make another pull request